packages/auth/src/core/util/handler.ts

```diff
@@ -107,7 +114,17 @@ export function _getRedirectUrl(
       delete paramsDict[key];
     }
   }
-  return `${getHandlerBase(auth)}?${querystring(paramsDict).slice(1)}`;
+
+  // Sets the App Check token to pass to the widget
+  const appCheckToken = await auth._getAppCheckToken();
+  const appCheckTokenFragment = appCheckToken
+    ? `${FIREBASE_APP_CHECK_FRAGMENT_ID}=${encodeURIComponent(appCheckToken)}`
+    : '';
+
+  // Start at index 1 to skip the leading '&' in the query string
+  return `${getHandlerBase(auth)}?${querystring(paramsDict).slice(
+    1
+  )}#${appCheckTokenFragment}`;
```

Comment:
nit: Does this add an empty fragment if the App Check token is not present? If so, can we not add the fragment at all?

Reply:
Yeah, I went with what was done with query params, where the ? is included regardless of if query params are present, but I think that makes more sense to not include the # at all if the fragment is empty. Updated!